### ISSUE TITLE
Inconsistent file/module names in CLMFatesParamInterfaceMod

### ISSUE THREAD

**Issue by apcraig** [ESCOMP/CTSM#951]
### Brief summary of bug

CLMFatesParamInterfaceMod is the module name in file clmfates_paraminterfaceMod.F90.  This is the only module in CLM that doesn't adhere to the filename/module name consistency standard.  As a result, some build dependency tools do not resolve the dependency correctly and the build fails.  

### General bug information

**CTSM version you are using:** [output of `git describe`]
>git describe
ctsm1.0.dev083

git log reports 5cc4ede4ffebf5cd

**Does this bug cause significantly incorrect results in the model's science?** [Yes / No]
No

**Configurations affected:** [Fill this in if known.]
Any model building with a version of the makdep tool that is unable to handle the inconsistent module/filename implementation.

### Details of bug

CLMFatesParamInterfaceMod is the module name in file clmfates_paraminterfaceMod.F90.  This is the only module in CLM that doesn't adhere to the filename/module name consistency standard.  As a result, some build dependency tools do not resolve the dependency correctly and the build fails.  The module should be renamed CLMFates_ParamInterfaceMod and the following files will need to be updated,

src/utils/clmfates_paraminterfaceMod.F90
src/main/pftconMod.F90
src/fates/main/FatesInterfaceMod.F90

The other option is to rename the filename or to create some sort of module rename implementation.

### Important details of your setup / configuration so we can reproduce the bug

Running with CTSM imported into CESM1.

### Important output or errors that show the problem

Compile will fail due to inconsistency in file/module names.

**Comment by billsacks** [comment]
We're in the midst of doing a massive file rename across the whole CTSM code base (#869). This issue will be resolved in the course of doing that. (See also https://github.com/negin513/ctsm_filenames/blob/master/ctsm_filenames.csv though that is still a work in progress.)

@apcraig is this a critical blocker for your work? If not, I'd like to close this issue and just get it fixed along with the big file name overhaul, if that's okay with you.

By the way: what dependency tool(s) are you using? I know it used to be the case that CESM's mkdepends tool required file and module names to be the same, but that limitation was removed 5-10 years ago, and thus we haven't been enforcing complete consistency between module and file names. Again, with the planned massive file renames, we should return to a state where the two are consistent, at least for now.

**Comment by billsacks** [comment]
@apcraig I found this in cime's current mkdepends tool:

```
# Modifications to Brian Eaton's original to relax the restrictions on
# source file name matching module name and only one module per source
# file.  Also added a new "-d depfile" option which allows an additional
# file to be added to every dependence.
#
#
#   Tom Henderson
#   Global Systems Division, NOAA/OAR
#   Mar 2011
```

You may want to just grab the latest mkdepends tool for your work.

**Comment by apcraig** [comment]
We are coupling CTSM in an older version of CESM, basically CESM1.2.  We are using the build scripts in that version.  This is not blocking me right now.  I have renamed the module myself and made changes in other places.  If it were multiple files, I might upgrade the build, but since it's just one file, I have decided to modify the code.  It is fine to close this Issue, just wanted to bring it to your attention since it seems like an easy modification that could benefit others as well.  Thanks.
